leaflet/index.d.ts

```diff
@@ -1419,7 +1419,7 @@ declare namespace L {
     }
 
     export interface IconOptions extends BaseIconOptions {
-        iconUrl: string;
+        iconUrl?: string;
```

Comment:
In this case, this is not optional. The problem you were seeing was probably due to a bug in a recent update I made. The fix #14750 was (finally) merged today. Please let me know if the issue persists in the latest version of the typing.